Prose/ProseLib/Sources/AuthenticationFeature/AuthenticationScreen.swift

```diff
@@ -15,24 +17,205 @@ public struct AuthenticationScreen: View {
+                        }
                 }
-                Button { actions.send(.loginButtonTapped) } label: {
-                    Text("Log into your account")
+                WithViewStore(self.store, removeDuplicates: { $0.password == $1.password }) { viewStore in
+                    SecureField("Enter your password…", text: viewStore.binding(\.$password))
+                        .onSubmit { actions.send(.submitTapped(.password)) }
+                        .textContentType(.password)
+                        .disableAutocorrection(true)
+                        .focused($focusedField, equals: .password)
+                }
+            }
+            .disabled(viewStore.isLoading)
+        }
+    }
+
+    private func chatAddressHelpButton() -> some View {
+        WithViewStore(
+            self.store,
+            removeDuplicates: { $0.popover == $1.popover }
+        ) { viewStore in
+            Button { actions.send(.showPopoverTapped(.chatAddress)) } label: {
+                Image(systemName: "questionmark")
+            }
+            .buttonStyle(CircleButtonStyle())
+            .popover(
+                unwrapping: viewStore.binding(\.$popover),
+                case: /State.Popover.chatAddress,
+                content: { _ in Self.chatAddressPopover() }
+            )
+        }
+    }
+
+    private func logInButton() -> some View {
+        WithViewStore(
+            self.store,
+            removeDuplicates: { $0.isFormValid == $1.isFormValid && $0.isLoading == $1.isLoading }
+        ) { viewStore in
+            Button {
+                actions.send(viewStore.isLoading ? .cancelLogInTapped : .loginButtonTapped)
+            } label: {
+                Text(viewStore.isLoading ? "Cancel" : "Log into your account")
+                    .frame(minWidth: 196)
+            }
+            .overlay(alignment: .leading) {
+                if viewStore.isLoading {
+                    ProgressView()
+                        .scaleEffect(0.5, anchor: .center)
                 }
-                .disabled(!viewStore.isFormValid)
             }
-            .padding(48)
+            .disabled(!viewStore.isFormValid && !viewStore.isLoading)
+        }
+    }
+
+    private func footer() -> some View {
+        HStack(spacing: 16) {
+            WithViewStore(
+                self.store,
+                removeDuplicates: { $0.popover == $1.popover }
+            ) { viewStore in
+                Button("Lost your password?") { actions.send(.showPopoverTapped(.passwordLost)) }
+                    .popover(
+                        unwrapping: viewStore.binding(\.$popover),
+                        case: /State.Popover.passwordLost,
+                        content: { _ in Self.passwordLostPopover() }
+                    )
+                Divider().frame(height: 18)
+                Button("No account yet?") { actions.send(.showPopoverTapped(.noAccount)) }
```

Comment:
Why don't you finish the localization when you're already building the UI and save us both another PR cycle?

Reply:
Ok, I will do that 🙂